compiler/haskell-ide-core/src/Development/IDE/UtilGHC.hs

```diff
@@ -106,22 +54,14 @@ prettyPrint = showSDoc fakeDynFlags . ppr
 
 runGhcEnv :: HscEnv -> Ghc a -> IO a
 runGhcEnv env act = do
-    ref <- newIORef env
-    unGhc act $ Session ref
-
+    filesToClean <- newIORef emptyFilesToClean
```

Comment:
Does this PR change the tempfiles we create or is this something unrelated that you included in this PR (both is fine with me, just trying to understand why we need this now).

Reply:
Unrelated. We should probably have had this from the beginning, but I don't think anything we actually do uses temp files, so it's somewhat irrelevant. However, if it ever becomes relevant, we'll get horrible race conditions that are impossible to debug, so I thought worth cleaning up now.